### ISSUE TITLE
EE edition O2/3 optimization causes at build time: Cannot invoke "org.graalvm.compiler.nodes.ValueNode.stamp(org.graalvm.compiler.nodes.NodeView)" because "nextValue" is null

### ISSUE THREAD

**Issue by CGDogan** [oracle/graal#7073]
Latest EE edition v20 on ARM linux

```
#18 4.394 [1/8] Initializing...                                                                                    (3.8s @ 0.21GB)
#18 4.394  Java version: 20.0.2+9, vendor version: Oracle GraalVM 20.0.2+9.1
#18 4.394  Graal compiler: optimization level: 2, target machine: armv8-a, PGO: ML-inferred
#18 4.395  C compiler: gcc (linux, aarch64, 12.2.0)
#18 4.395  Garbage collector: Serial GC (max heap size: 80% of RAM)
#18 46.01 [2/8] Performing analysis...  [*****]                                                                   (41.5s @ 1.88GB)
#18 46.02   14,935 (88.00%) of 16,971 types reachable
#18 46.02   33,183 (68.29%) of 48,594 fields reachable
#18 46.06   91,572 (62.83%) of 145,750 methods reachable
#18 46.06    4,512 types,   267 fields, and 1,608 methods registered for reflection
#18 46.06       77 types,   101 fields, and    78 methods registered for JNI access
#18 46.06        4 native libraries: dl, pthread, rt, z
#18 51.47 [3/8] Building universe...                                                                               (5.4s @ 2.05GB)
#18 68.84 [4/8] Parsing methods...      [****]                                                                    (17.2s @ 1.60GB)
#18 70.74 [5/8] Inlining methods...     [***]                                                                      (1.6s @ 2.05GB)
#18 238.0 [6/8] Compiling methods...    [*************]                                                          (167.3s @ 2.04GB)
#18 238.0 
#18 238.0 Fatal error: org.graalvm.compiler.debug.GraalError: java.lang.NullPointerException: Cannot invoke "org.graalvm.compiler.nodes.ValueNode.stamp(org.graalvm.compiler.nodes.NodeView)" because "nextValue" is null
#18 238.0 	at method: XMPDateTime com.adobe.internal.xmp.impl.ISO8601Converter.parse(String, XMPDateTime)  [Direct call from XMPDateTime XMPUtils.convertToDate(String)]
#18 238.0 	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:1297)
#18 238.0 	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue.doCompile(CompileQueue.java:1226)
#18 238.0 	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue$CompileTask.run(CompileQueue.java:300)
#18 238.0 	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:187)
#18 238.0 	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:171)
#18 238.0 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
#18 238.0 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
#18 238.0 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
#18 238.0 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
#18 238.0 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
#18 238.0 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
#18 238.0 Caused by: java.lang.NullPointerException: Cannot invoke "org.graalvm.compiler.nodes.ValueNode.stamp(org.graalvm.compiler.nodes.NodeView)" because "nextValue" is null
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.type.StampTool.meetOrNull(StampTool.java:69)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.ValuePhiNode.inferStamp(ValuePhiNode.java:84)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.tryInferStamp(CanonicalizerPhase.java:702)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.processNode(CanonicalizerPhase.java:467)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.processWorkSet(CanonicalizerPhase.java:315)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:76)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:44)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.applyIncremental(CanonicalizerPhase.java:237)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase$Apply.close(IncrementalCanonicalizerPhase.java:112)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:441)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.PhaseSuite.run(PhaseSuite.java:390)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.core.phases.BaseTier.run(BaseTier.java:47)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.emitFrontEnd(GraalCompiler.java:300)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:169)
#18 238.0 	at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:137)
#18 238.0 	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:1278)
#18 238.0 	... 10 more
#18 238.0 ------------------------------------------------------------------------------------------------------------------------
#18 238.0                        20.0s (8.4% of total time) in 280 GCs | Peak RSS: 4.53GB | CPU load: 5.76
#18 238.0 ========================================================================================================================
#18 238.0 Finished generating 'libbfbridge' in 3m 57s.
#18 ERROR: process "/bin/sh -c native-image -Djava.awt.headless=true -cp \".:jar_files/*\" --shared -H:Name=libbfbridge   org.camicroscope.BFBridge" did not complete successfully: exit code: 1

```

 the relevant source is https://github.com/drewnoakes/adobe-xmp-core/blob/master/com/adobe/internal/xmp/impl/ISO8601Converter.java#L90

## Reproduce

git clone https://github.com/CGDogan/adobe-xmp-core.git

cd adobe-xmp-core

javac $(find . -name "*.java")

native-image com.adobe.internal.xmp.XMPUtils  -O2

**Comment by knuspertante** [comment]
Same for version 17

`
GraalVM Runtime Environment Oracle GraalVM 17.0.8+9.1 (build 17.0.8+9-LTS-jvmci-23.0-b14)
Substrate VM Oracle GraalVM 17.0.8+9.1 (build 17.0.8+9-LTS, serial gc, compressed references)
`

```shell
[6/8] Compiling methods...    [****************]                                                       (273,7s @ 4,33GB)
Fatal error: org.graalvm.compiler.debug.GraalError: java.lang.NullPointerException: Cannot invoke "org.graalvm.compiler.nodes.ValueNode.stamp(org.graalvm.compiler.nodes.NodeView)" because "nextValue" is null
	at method: XMPDateTime com.adobe.internal.xmp.impl.ISO8601Converter.parse(String, XMPDateTime)  [Direct call from XMPDateTime XMPUtils.convertToDate(String)]
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:1297)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue.doCompile(CompileQueue.java:1226)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue$CompileTask.run(CompileQueue.java:300)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:187)
	at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:171)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException: Cannot invoke "org.graalvm.compiler.nodes.ValueNode.stamp(org.graalvm.compiler.nodes.NodeView)" because "nextValue" is null
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.type.StampTool.meetOrNull(StampTool.java:69)
	at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.ValuePhiNode.inferStamp(ValuePhiNode.java:84)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.tryInferStamp(CanonicalizerPhase.java:702)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.processNode(CanonicalizerPhase.java:467)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.processWorkSet(CanonicalizerPhase.java:315)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:76)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase.run(IncrementalCanonicalizerPhase.java:44)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.CanonicalizerPhase.applyIncremental(CanonicalizerPhase.java:237)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.common.IncrementalCanonicalizerPhase$Apply.close(IncrementalCanonicalizerPhase.java:112)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:441)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.PhaseSuite.run(PhaseSuite.java:390)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
	at jdk.internal.vm.compiler/org.graalvm.compiler.core.phases.BaseTier.run(BaseTier.java:47)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:434)
	at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:322)
	at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.emitFrontEnd(GraalCompiler.java:300)
	at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:169)
	at jdk.internal.vm.compiler/org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:137)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.code.CompileQueue.defaultCompileFunction(CompileQueue.java:1278)
	... 10 more
```

**Comment by oubidar-Abderrahim** [comment]
Thank you for reporting this we'll take a look into it shortly

**Comment by zingermax** [comment]
any updates/workaround for this issue?

**Comment by rmosaner** [comment]
Hi,

I just reproduced the bug on your version of Graal. 
In the upcoming release, the crash seems to not occur anymore. I will still verify that the changes since the last release really addressed the bug properly.

**Comment by rmosaner** [comment]
The root cause for the problem has been found and will be fixed in the upcoming release.
In the meantime, you can try to run your application with native-image flag `-H:-VectorizeLoops` (disclaimer: this can reduce performance)

**Comment by oubidar-Abderrahim** [comment]
Fixed in 23.1.0
